types/index.d.ts

```diff
@@ -0,0 +1,21 @@
+import { type FunctionComponentElement, type ProviderProps, type PropsWithChildren } from "react";
+
+interface Options {
+  debounce?: boolean,
+  events?: Record<string, Array<string>>
+  modelTypes?: Record<string, object>,
+  withRef?: boolean
+}
+
+export declare function connectBackboneToReact<Model extends {}, ModelProps extends {}>(
+  mapModelToProps: (models: Model, props: ModelProps) => Partial<ModelProps>, options?: Options
+): <CombinedProps>(
+  wrappedComponent: React.ComponentType<CombinedProps>
+) => React.ComponentType<Omit<CombinedProps, keyof ModelProps>>;
```

Comment:
I think `ModelProps` might be inconsistent here. As the second param to `mapModelsToProps`, it should me all of the props passed in from the parent component. But then in the return type you're omitting those props from what needs to be passed in from the parent component.

Compared to what I had written in the JIRA ticket:

```tsx
function connectBackboneToReact<Models extends {}, ModelProps extends {}>(
  mapModelsToProps: (models: Models) => ModelProps
): <CombinedProps>(
  WrappedComponent: React.ComponentType<CombinedProps>
) => React.ComponentType<Omit<CombinedProps, keyof ModelProps>>;
```

In that definition, `ModelProps` was the props specifically being added by `mapModelsToProps`, so omitting them from the generated component type meant that the parent wouldn't have to pass in anything that was derived from a backbone model. I didn't have a type included for the `props` param though, so we probably can't just use that as-is. I'm not 100% sure what the best way to write that is where TS can still infer everything properly, tbh.